core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -558,4 +558,32 @@ default ClientRequestContext unwrap() {
     default ClientRequestContext unwrapAll() {
         return (ClientRequestContext) RequestContext.super.unwrapAll();
     }
+
+    /**
+     * Returns the authority which will eventually be sent when a {@link Client} sends an {@link HttpRequest}.
+     * This method checks the following locations and returns the first non-null value.
+     * <ol>
+     *     <li>Either the {@link HttpHeaderNames#HOST} or {@link HttpHeaderNames#AUTHORITY} value from
+     *         {@link ClientRequestContext#additionalRequestHeaders()}.</li>
+     *     <li>The {@link HttpRequest#authority()} from {@link ClientRequestContext#request()}.</li>
+     *     <li>{@link ClientRequestContext#defaultRequestHeaders()}.</li>
+     *     <li>{@link Endpoint#authority()}.</li>
+     * </ol>
+     */
+    @Nullable
+    String authority();
```

Comment:
Could we move these methods so they are closer to URI-related getters, such as `fragment()`?